File: Exercism/dart/triangle/lib/triangle.dart
class Triangle {
  bool equilateral(double a, double b, double c) {
    return a == b && b == c;
  }

  bool isosceles(double a, double b, double c) {
    return a == b || b == c || a == c;
  }

  bool scalene(double a, double b, double c) {
    return a != b && b != c && a != c;
  }
}

File: Exercism/ruby/triangle/triangle.rb
# frozen_string_literal: true

# Triangle ..
class Triangle
  def initialize(input)
    @a = input[0]
    @b = input[1]
    @c = input[2]
  end

  def triangle?
    !(@a <= 0 || @b <= 0 || @c <= 0 ||
      @a + @b < @c || @b + @c < @a || @c + @a < @b)
  end

  def equilateral?
    return false unless triangle?
    
    @a == @b && @b == @c
  end

  def isosceles?
    return false unless triangle?

    @a == @b || @b == @c || @a == @c
  end

  def scalene?
    return false unless triangle?

    @a != @b && @b != @c && @a != @c
  end
end
